part3/roles-responsibilities.qmd

```diff
@@ -2,13 +2,220 @@
+In this section, we will explore key roles and responsabilities within GenAI projects.
 
 ## Key Roles
 
-- Agent Engineers
-- MLOps Engineers
-- Data Scientists
-- Platform Engineers
+Most organisations are consumers of AI models rather than builders. Their primary focus tends to be on integration and orchestration, agent systems, cost management, prompt engineering, and ensuring safety and governance.
 
-*Content to be developed*
\ No newline at end of file
+The key roles listed below relate to how organisations implement GenAI. Because GenAI work requires significant collaboration, the exact division of responsibilities often depends on the organisation’s size. In some cases, one person may take on several roles, or hold a hybrid role that combines responsibilities from different areas.
+
+### Technical Roles
+
+These roles directly build, implement, and maintain AI systems, handling the hands-on technical work and domain expertise required to deliver AI solutions.
+
+#### AI Engineer {.unnumbered}
+An AI Engineer integrates and optimizes pre-trained AI models into production applications, focusing on making AI capabilities work reliably and cost-effectively within business systems rather than building models from scratch.
+
+Primary Responsibilities:
+
+* Integrate and fine-tune existing LLMs with most organizations using pre-trained models via APIs
+* Design and implement RAG (Retrieval Augmented Generation) architectures for domain specific applications
+* Develop prompt engineering strategies and optimize prompt templates for production use
+* Build evaluation and testing frameworks for non-deterministic AI systems, ensuring quality despite variable outputs
+* Optimize for cost and performance on the application logic level using prompt optimization, model selection, workflow design that minimizes API calls and output optimizaion.
+* Create AI agents workflows and orchestrate AI tool chains for complex tasks
+* Deploy AI models into production environments with proper versioning and monitoring
+
+#### Data Engineer {.unnumbered}
+GenAI Data Engineers support all data needs for AI applications. They're the bridge between raw data sources and AI systems, whether that's for RAG, AI agents, fine-tuning, analytics, or traditional business intelligence.
+
+Primary Responsibilities:
+
+* Create data pipelines that inject structured business data into AI prompts (customer records, product catalogs, transaction history)
+* Process unstructured documents for AI consumption
+* Handle multi-modal data preprocessing (images, audio, video) for specialized AI applications
+* Build and maintain vector databases for semantic search and RAG applications
+* Implement embedding pipelines using embedding models
```

Comment:
This part of vector database creation. Should we include it on the same line or as a sub-bullet?